hack/run-functional-tests.sh

```diff
@@ -73,107 +73,171 @@ check_arguments() {
+  # define local index from second argument
+  local resource_index="${2}"
 
-    if [ "${manifest_list}" == "" ]; then
+  # grab local resource from ${TESTS_MANIFEST}
+  local resource
+  
+  resource=$(yq read "${TESTS_MANIFEST}" "resources.[${resource_index}].kind")
 
-      echo "[WARN] No \"${test_type}\" tests for \"${resource}\". Skipping..."
-      echo "============================================================================"
+  # grab local test type from ${TESTS_MANIFEST}
+  local test_type
+  
+  test_type=$(yq read "${TESTS_MANIFEST}" "resources.[${resource_index}].desired")
+
+  # grab local list of test manifests to use
+  local manifest_list
+  
+  manifest_list=$(yq read -P "${TESTS_MANIFEST}" "resources.[${resource_index}].manifests" | sed 's/^-[ ]*//')
+
+  # grab local user_script specified for pre/post/between running
+  local user_script
+  
+  user_script=$(yq read -P "${TESTS_MANIFEST}" "resources.[${resource_index}].script")
+
+  # full path to the user specified script associate with this stanza in the manifest
+  local user_script_path="testing/${resource}/scripts/${user_script}"
+
+  if [ "${manifest_list}" == "" ]; then
+
+    echo "[WARN] No \"${test_type}\" tests for \"${resource}\". Skipping..."
+    echo "============================================================================"
+
+  else
+
+    echo "[INFO] **** Running \"${test_type}\" tests for \"${resource}\" ****"
+    echo "============================================================================"
 
-    else
+    # check to see if the user specified a script to be associated with this stanza
+    # only run script for apply actions
+    if [[ ${user_script} != "" ]] && [[ "${action}" == "apply" ]]; then 
+
+      # if they did specify a script run it with the setup argument and pass the namespace in
+      ${user_script_path} "setup" "${TEST_NAMESPACE}"
 
-      echo "[INFO] **** Running \"${test_type}\" tests for \"${resource}\" ****"
       echo "============================================================================"
 
-      for testfile in ${manifest_list}; do
+    fi
+
+    for testfile in ${manifest_list}; do
 
-          local test_file_path="testing/${resource}/${testfile}"
+        local test_file_path="testing/${resource}/${testfile}"
 
-          if [ -f "${test_file_path}" ]; then
+        if [ -f "${test_file_path}" ]; then
 
-              echo "[INFO] ${action}: \"${testfile}\""
+            echo "[INFO] ${action}: \"${testfile}\""
+            
+            if [ "${action}" == "delete" ]; then
               
+              # kubectl doesn't like double quites here.
+              # disable checking for double quotes around variables.
+              # shellcheck disable=SC2086
+              kubectl ${action} -f "${test_file_path}" -n ${TEST_NAMESPACE} --ignore-not-found
+
+            else
+
+              # kubectl doesn't like double quites here. disable checking for double quotes around variables
```

Comment:
```suggestion
              # kubectl doesn't like double quotes here. disable checking for double quotes around variables
```

Also, Nit on consistency with one line vs. 2 line comment as on line 137/138